### ISSUE TITLE
Crash in `GodotNavigationServer::map_get_path`

### ISSUE THREAD

**Issue by Scony** [godotengine/godot#60413]
### Godot version

3.x (3ba980379dfb72f5d238640835d8e77c76cf3c99) and master

### System information

manjaro linux

### Issue description

This is a regression introduced in c52d842435325f0f244177053c3c6c84d69dcf72 via https://github.com/godotengine/godot/pull/60242 (reverting resolves the issue)

I get such a crash on one of my demos:
```
Program terminated with signal SIGABRT, Aborted.
#0  0x00007fcd4c53ed22 in raise () from /usr/lib/libc.so.6
[Current thread is 1 (Thread 0x7fcd4c3d07c0 (LWP 15219))]
(gdb) bt
#0  0x00007fcd4c53ed22 in raise () from /usr/lib/libc.so.6
#1  0x00007fcd4c528862 in abort () from /usr/lib/libc.so.6
#2  0x00000000008059ce in handle_crash (sig=<optimized out>) at platform/x11/crash_handler_x11.cpp:51
#3  <signal handler called>
#4  NavMap::get_path (this=0x160a3260, p_origin=..., p_destination=..., p_optimize=<optimized out>) at modules/navigation/nav_map.cpp:164
#5  0x0000000000dc6f52 in GodotNavigationServer::map_get_path (this=<optimized out>, p_map=..., p_origin=..., p_destination=..., p_optimize=<optimized out>) at modules/navigation/godot_navigation_server.cpp:222
#6  0x0000000002c935f3 in Navigation2DServer::map_get_path (this=<optimized out>, p_map=..., p_origin=..., p_destination=..., p_optimize=<optimized out>) at servers/navigation_2d_server.cpp:186
#7  0x0000000002753297 in Navigation2D::get_simple_path (this=this@entry=0x160a2dd0, p_start=..., p_end=..., p_optimize=p_optimize@entry=false) at scene/2d/navigation_2d.cpp:74
#8  0x000000000275660a in MethodBind3RC<Vector<Vector2>, Vector2 const&, Vector2 const&, bool>::call (this=0x8860b60, p_object=<optimized out>, p_args=0x7ffcd0cd4440, p_arg_count=-791853952, r_error=...) at ./core/method_bind.gen.inc:2849
#9  0x000000000328052a in Object::call (this=0x160a2dd0, p_method=..., p_args=0x16024a48, p_argcount=3, r_error=...) at core/object.cpp:918
#10 0x00000000033272c1 in Variant::call_ptr (this=this@entry=0x16024988, p_method=..., p_args=p_args@entry=0x16024a48, p_argcount=p_argcount@entry=3, r_ret=0x16024a00, r_error=...) at core/variant_call.cpp:1185
#11 0x000000000093b659 in GDScriptFunction::call (this=0x1607fc00, p_instance=0x8ff2160, p_args=<optimized out>, p_argcount=<optimized out>, r_err=..., p_state=<optimized out>) at modules/gdscript/gdscript_function.cpp:1050
#12 0x00000000009543bf in GDScriptFunctionState::resume (this=0x16073a30, p_arg=...) at modules/gdscript/gdscript_function.cpp:1814
#13 0x0000000000955395 in GDScriptFunctionState::_signal_callback (this=0x16073a30, p_args=0x16787380, p_argcount=1, r_error=...) at modules/gdscript/gdscript_function.cpp:1762
#14 0x0000000000957314 in MethodBindVarArg<GDScriptFunctionState>::call (this=<optimized out>, p_object=<optimized out>, p_args=<optimized out>, p_arg_count=<optimized out>, r_error=...) at ./core/method_bind.h:320
#15 0x000000000328052a in Object::call (this=0x16073a30, p_method=..., p_args=0x16787380, p_argcount=1, r_error=...) at core/object.cpp:918
#16 0x000000000328b6d9 in Object::emit_signal (this=0x8a9ed30, p_name=..., p_args=<optimized out>, p_argcount=<optimized out>) at core/object.cpp:1231
#17 0x000000000328c923 in Object::emit_signal (this=this@entry=0x8a9ed30, p_name=..., p_arg1=..., p_arg2=..., p_arg3=..., p_arg4=..., p_arg5=..., p_arg6=..., p_arg7=..., p_arg8=...) at core/object.cpp:1286
#18 0x000000000220be2e in SceneTree::idle (this=0x8a9ed30, p_time=0.0166666675) at scene/main/scene_tree.cpp:589
#19 0x0000000000833984 in Main::iteration () at main/main.cpp:2271
#20 0x0000000000816b21 in OS_X11::run (this=this@entry=0x7ffcd0cd52e0) at platform/x11/os_x11.cpp:3842
#21 0x00000000007fdd1f in main (argc=11, argv=0x7ffcd0cd5bf8) at platform/x11/godot_x11.cpp:55
```

CC @kleonc 

### Steps to reproduce

Run `2d/GetSimplePathTests.tscn` demo from MRP

### Minimal reproduction project

[godot-navigation-demos.zip](https://github.com/godotengine/godot/files/8534182/godot-navigation-demos.zip)

**Comment by kleonc** [comment]
@Scony Can't reproduce, it's not crashing for me on Windows (`"platform=windows", "target=debug", "bits=64", "tools=yes"`). I looked at it again and I don't see anything incorrect/wrong in #60242. The crash at line 164. is unexpected to me:
https://github.com/godotengine/godot/blob/3ba980379dfb72f5d238640835d8e77c76cf3c99/modules/navigation/nav_map.cpp#L163-L165
Really not sure what's the issue here. 
The only suspicious thing I noticed is that across the navigation code `NULL` is used instead of `nullptr` but I somehow doubt it has anything to do with this crash.
Or maybe I'm just blind and there's indeed something wrong with my fix. :smile: 

Also does #60241 results in the similar crash for you in `master`?
